### PR TITLE
[sfp-refactor] Add initial support for SFF-8472 in sonic_xcvr

### PR DESCRIPTION
**Pull request by andywongarista**
#### Description

This PR adds basic support for the SFF-8472 transceiver spec within the new sonic_xcvr package.

#### Motivation and Context

See the HLD: [Azure/SONiC#786](https://github.com/Azure/SONiC/pull/786)

#### How Has This Been Tested?

Added new unit test under tests/sonic_xcvr
Verified API calls on an Arista platform

### CODE REVIEW

File: sonic_platform_base/sonic_xcvr/fields/public/sff8472.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import struct
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/fields/public/sff8472.py:1]: Can you add unit test for these new fields?

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@andywongarista please fix the build errors due to your changes:

```
self = <tests.sonic_xcvr.test_xcvr_field.TestDateField object at 0x7f92f90350d0>

    def test_decode(self):
        field = mem_map.get_field("Date")
        data = bytearray(b'\x32\x31\x31\x30\x32\x30\x00')
>       decoded = field.decode(data)

tests/sonic_xcvr/test_xcvr_field.py:234: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sonic_platform_base/sonic_xcvr/fields/xcvr_field.py:247: in decode
    date = super(DateField, self).decode(raw_data, **decoded_deps)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sonic_platform_base.sonic_xcvr.fields.xcvr_field.DateField object at 0x7f92f9182890>
raw_data = bytearray(b'211020\x00'), decoded_deps = {}

    def decode(self, raw_data, **decoded_deps):
>       return struct.unpack(self.format, raw_data)[0].decode(self.encoding)
E       error: unpack requires a string argument of length 8

sonic_platform_base/sonic_xcvr/fields/xcvr_field.py:175: error
```

**Comment by prgeor** [comment]
@andywongarista i think sonic-mgmt tests for the API set_power_override() will fail since its not applicable for SFF-8472 [https://github.com/Azure/sonic-mgmt/blob/master/tests/platform_tests/api/test_sfp.py#L516](https://github.com/Azure/sonic-mgmt/blob/master/tests/platform_tests/api/test_sfp.py#L516)

**Comment by andywongarista** [comment]
> @andywongarista i think sonic-mgmt tests for the API set_power_override() will fail since its not applicable for SFF-8472 https://github.com/Azure/sonic-mgmt/blob/master/tests/platform_tests/api/test_sfp.py#L516

https://github.com/Azure/sonic-mgmt/pull/4668

**Comment by prgeor** [comment]
@andywongarista following tests will fail for SFP, QSFP+, QSFP28 for platforms using SfpOptoeBase

1. test_reset()
2. test_tx_disable()
3. test_tx_disable_channel()
4. test_lpmode()
5. test_power_override()

Will you be able to fix this in the sfp-refactor code

**Comment by prgeor** [comment]
@andywongarista could you resolve merge conflict?

**Comment by prgeor** [comment]
@andywongarista could you resolve merge conflict?

**Review by prgeor — APPROVED** [review]

**Pull request merged**